porcupine/plugins/python_venv.py

```diff
@@ -109,25 +109,24 @@ def _on_folder_refreshed(event: utils.EventWithData) -> None:
             tree.tk.call(tree, "tag", "add", "venv", [venv_id])
 
 
-def _on_treeview_right_click(event: tkinter.Event[dirtree.DirectoryTree]) -> str:
-    tree = event.widget
-    tree.tk.call("focus", tree)
-
-    item: str = tree.identify_row(event.y)  # type: ignore[no-untyped-call]
-    tree.set_the_selection_correctly(item)
-
+def _populate_menu(event: tkinter.Event[dirtree.DirectoryTree]) -> None:
+    tree: dirtree.DirectoryTree = event.widget
+    [item] = tree.selection()
     path = dirtree.get_path(item)
     project_root = dirtree.get_path(tree.find_project_id(item))
-    if is_venv(path) and get_venv(project_root) != path:
-        menu = tkinter.Menu(tearoff=False)
-        menu.add_command(
-            label="Use this Python venv",
-            # No need to refresh when clicked, somehow already refreshes 4 times (lol)
-            command=partial(set_venv, project_root, path),
-        )
-        menu.tk_popup(event.x_root, event.y_root)
-        menu.bind("<Unmap>", (lambda event: menu.after_idle(menu.destroy)), add=True)
-    return "break"
+    if path == project_root or not path.is_dir():
+        return
+
+    is_used_var = tkinter.BooleanVar(value=(get_venv(project_root) == path))
+    cast(Any, tree.contextmenu).garbage_collection_is_lol = is_used_var
+
+    tree.contextmenu.add_checkbutton(
+        label="Use this Python venv",
+        variable=is_used_var,
+        state=("normal" if is_venv(path) else "disabled"),
```

Comment:
Should be also disabled whenever it is checked. Otherwise user thinks that clicking it will somehow stop using the venv, and that's just not true.